File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.py
class Solution:
    def numberOfArrays(self, d: List[int], lower: int, upper: int) -> int:
        n = len(d)
        n += 1
        prev = 1
        mn, mx = 1, 1

        for i in range(1,n):
            prev = (prev + d[i-1])
            mn = min(mn, prev)
            mx = max(mx, prev)

        diff = -(mn-lower)
        high = mx + diff
        res = upper - high + 1

        if res < 0:
            return 0

        return res 

File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.cpp
class Solution {
public:
    int numberOfArrays(vector<int>& diff, int lower, int upper) {
        int n = diff.size(); n++;
        
        long mn = 1, mx = 1, prev = 1;

        for (int i = 1; i < n; i++) {
            prev = (prev + diff[i-1]);
            mn = min(mn, prev);
            mx = max(mx, prev);
        }

        long high = mx - (mn-lower);
        long res = upper - high + 1;

        if (res < 0)
            return 0;

        return res; 
    }
};